### PR TITLE
feat: Add more lint rules

### PR DESCRIPTION
**Pull request by spydon**
# Description

Adds some more lint rules and fixes the issues those rules pointed out.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

## Breaking Change

- [x] No, this is *not* a breaking change.

## Related Issues

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Review by st-pasha — CHANGES_REQUESTED** [review]
I just tried adding a single rule
```
linter:
  rules:
    - prefer_asserts_with_message
```
Into the `packages/flame/analysis_options.yaml` file, and sure enough it reports 28 violations of that rule. Which means that even though this PR adds the same rule, the test system somehow failed to run the proper analysis with that rule.

I haven't tried with any other rule, but I suspect there are 100 or so violations still hiding in the larger repo, and we just need to figure out why they are hiding...

File: examples/games/trex/pubspec.yaml [hunk 1]
```diff
@@ -11,9 +11,9 @@ environment:
 
 dependencies:
   flame: ^1.1.1
-  google_fonts: ^2.3.2
   flutter:
     sdk: flutter
+  google_fonts: ^2.3.2
```
Comment (erickzanardo) [inline examples/games/trex/pubspec.yaml:16]: Not need to do now, but I believe this isn't used anymore? I believe Pasha has changed this to use Sprite fonts.

Comment (spydon) [reply examples/games/trex/pubspec.yaml:16]: Removed

File: packages/flame_bloc/lib/src/flame_bloc_game.dart [hunk 2]
```diff
@@ -138,5 +138,6 @@ mixin FlameBloc on FlameGame {
 class FlameBlocGame extends FlameGame with FlameBloc {
   /// FlameBlocGame constructor with an optional [Camera] as a parameter to
   /// FlameGame.
+  @Deprecated('Use FlameBlocProvider and FlameBlocListener instead')
```
[the remark this answers is not in the document]
Comment (erickzanardo) [reply packages/flame_bloc/lib/src/flame_bloc_game.dart:141]: This should be on its own PR though?

**Comment by spydon** [comment]
> I just tried adding a single rule
> 
> ```
> linter:
>   rules:
>     - prefer_asserts_with_message
> ```
> 
> Into the `packages/flame/analysis_options.yaml` file, and sure enough it reports 28 violations of that rule. Which means that even though this PR adds the same rule, the test system somehow failed to run the proper analysis with that rule.
> 
> I haven't tried with any other rule, but I suspect there are 100 or so violations still hiding in the larger repo, and we just need to figure out why they are hiding...

I think I broke it by forgetting to add `rules:` on the Flame pubspec, fixing now.

File: packages/flame_bloc/lib/src/flame_bloc_game.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (spydon) [reply packages/flame_bloc/lib/src/flame_bloc_game.dart:141]: It is already deprecated, this just puts it on the constructor too (before it was only on the class).

[the remark this answers is not in the document]
Comment (erickzanardo) [reply packages/flame_bloc/lib/src/flame_bloc_game.dart:141]: got it!

**Review by erickzanardo — APPROVED** [review]

**Review by st-pasha — APPROVED** [review]

**Pull request merged**